src/hotspot/os/windows/gc/z/zVirtualMemory_windows.cpp

```diff
@@ -139,7 +139,7 @@ class ZVirtualMemoryManagerSmallPages : public ZVirtualMemoryManagerImpl {
   virtual bool reserve(uintptr_t addr, size_t size) {
     const uintptr_t res = ZMapper::reserve(addr, size);
 
-    assert(res == addr || res == nullptr, "Should not reserve other memory than requested");
+    assert(res == addr || res == reinterpret_cast<uintptr_t>(nullptr), "Should not reserve other memory than requested");
```

Comment:
Seems to me if the returned value is uintptr_t then this should be a check against zero not NULL/nullptr.

Reply:
Yes, please change this to `res == 0`.